### PR TITLE
change mode when schema is specified

### PR DESCRIPTION
**Pull request by starry97**
It works, but throws an error in monaco editor
screen shot 2017-07-11 at 4 32 38 pm
Seems like a bug in monaco editor?

### CODE REVIEW

**Comment by domoritz** [comment]
The error may happen because we pass an empty spec to the editor. Can you look what specs we are passing to the editor?

**Comment by domoritz** [comment]
Looks good but won't merge until the error above is resolved.

**Comment by starry97** [comment]
@domoritz The spec is not empty when the error is thrown

File: src/components/input-panel/spec-editor/index.js [hunk 1]
```diff
@@ -29,6 +30,16 @@ const mapDispatchToProps = function(dispatch) {
 
     updateEditorString: (val) => {
       dispatch(EditorActions.updateEditorString(val));
+    },
+
+    updateVegaSpecInVegaLite: (spec) => {
+      dispatch(EditorActions.changeMode(MODES.Vega));
```
Comment (domoritz) [inline src/components/input-panel/spec-editor/index.js]: If this just changes the mode, why not change `updateVegaSpec` to change the mode to Vega?

File: src/components/input-panel/spec-editor/renderer.js [hunk 2]
```diff
@@ -86,6 +79,23 @@ export default class Editor extends React.Component {
...
+    let schema, parsedMode;
+    try {
+      schema = JSON.parse(spec).$schema;
+    } catch (err) {
+      console.warn('Error parsing json string');
+    } 
+    if (schema) {
+      parsedMode = parser(schema).library;
+    }  
+    if (parsedMode === MODES.Vega || (!parsedMode && this.props.mode === MODES.Vega)) {
```
Comment (domoritz) [inline src/components/input-panel/spec-editor/renderer.js:93]: Can you merge this if with the one above?

Comment (starry97) [reply src/components/input-panel/spec-editor/renderer.js:93]: @domoritz You mean change it to 
`
if ((schema && parser(schema).library === MODES.Vega) || (!parser(schema).library && this.props.mode === MODES.Vega)) 
`?

Comment (domoritz) [reply src/components/input-panel/spec-editor/renderer.js:93]: or 

```
if (schema) {
parsedMode = parser(schema).library;
if (parsedMode === MODES.Vega || (!parsedMode && this.props.mode === MODES.Vega)) {
...
}
}
```

Comment (starry97) [reply src/components/input-panel/spec-editor/renderer.js:93]: @domoritz but then we need to have more lines to handle the case when schema is not defined

Comment (domoritz) [reply src/components/input-panel/spec-editor/renderer.js:93]: Fair.

**Pull request merged**